packages/flutter_test/lib/src/binding.dart

```diff
@@ -700,13 +727,40 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     // The LiveTestWidgetsFlutterBinding overrides this to report the exception to the console.
   }
 
+  Future<dynamic> _handleMessage(dynamic mockMessage) async {
+    final Map<dynamic, dynamic> data =
+        (mockMessage as Map<dynamic, dynamic>)['data'] as Map<dynamic, dynamic>;
```

Comment:
I don't remember, are the keys in the map really `dynamic`? I recall them all being `String`s.

Reply:
I tried both
`final Map<String, dynamic> data =
        (mockMessage as Map<dynamic, dynamic>)['data'] as Map<String, dynamic>;`
and
`final Map<String, dynamic> data =
        (mockMessage as Map<String, dynamic>)['data'] as Map<String, dynamic>;`

In both cases I get a casting error:
`the following _CastError was thrown running a test:
type '_InternalLinkedHashMap<Object?, Object?>' is not a subtype of type 'Map<String, dynamic>' in
type cast`

Is there a better way to cast the `mockMessage` and get the values inside `data`?

Reply:
you can use .cast<K, V> to cast to a certain type https://api.flutter.dev/flutter/dart-core/Map/cast.html

Reply:
Oh, interesting. In that case it's fine to keep it dynamic. @chunhtai's suggestion also works.

Reply:
Both `mockMessage` and `mockMessage['data']` are dynamic. It seems like `cast<K,V>()` is defined for Map and there isn't a cast function defined on dynamic type.

Reply:
@nbayati you can call any method on `dynamic`. So you can do `mockMessage.cast<String, dynamic>()` and it should work as long as `mockMessage` actually has that method.